contrib/jemalloc/include/jemalloc/internal/rb.h

```diff
@@ -43,15 +43,12 @@
 
 #ifdef __CHERI_PURE_CAPABILITY__
 #include <cheri/cheric.h>
-#define je_get_low_bits(ptr, mask)	cheri_get_low_ptr_bits(ptr, mask)
-#define je_clear_low_bits(ptr, mask)	cheri_clear_low_ptr_bits(ptr, mask)
-#define je_set_low_bits(ptr, bits)	cheri_set_low_ptr_bits(ptr, bits)
-#else
-#define je_get_low_bits(ptr, mask)	((uintptr_t)(ptr) & ((uintptr_t)(mask)))
-#define je_clear_low_bits(ptr, mask)	((uintptr_t)(ptr) & (~(uintptr_t)(mask)))
-#define je_set_low_bits(ptr, bits)	((uintptr_t)(ptr) | ((uintptr_t)(bits)))
 #endif
 
+#define je_get_low_bits(ptr, mask)	((uintptr_t)(ptr) & ((size_t)(mask)))
```

Comment:
I feel like there's an annotation I could apply to the mask casts rather than doing this type change but I can't remember/find it.

Reply:
`__attribute__((cheri_no_provenance))` (see clang/test/Sema/cheri/cap-provenance-attr.c)